src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -366,6 +370,22 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema, go
     }
   }
 
+  // ask Rawls if we have permission on this workspace. For writes, Rawls will also check if the workspace is locked.
+  private def validateUpsertPermissions(userInfo: UserInfo, workspaceNamespace: String, workspaceName: String, action: String = "read")(op: Boolean => Future[PerRequestMessage]) = {
+    val pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]] = authHeaders(requestContext) ~> sendRec
+
+    def enc(s: String) = java.net.URLEncoder.encode(s, "utf-8")
+
+    val checkUrl = s"${Rawls.workspacesUrl}/${enc(workspaceNamespace)}/${enc(workspaceName)}/checkIamActionWithLock/$action"
+
+    userAuthedRequest(Get(checkUrl))(userInfo: UserInfo) flatMap {
+      case resp if resp.status == NoContent => op(true)
+      case resp =>
+        logger.warn(s"****** workspace permissions failed with ${resp.status.intValue} ${resp.status.defaultMessage}: ${resp}")
```

Comment:
nit: why all the asterisks? doesn't this just lead to asterisk escalation in logs?

Reply:
thank you for this catch; I hadn't meant to leave that debugging log in.